### PR TITLE
vscode lsp support

### LINKED ISSUE #57: VSCode Extension for Import Linting
It would be great if there was a VSCode extension to help with live import linting, similar to what Pylance or the Ruff LSP can do. This way boundaries being broken can be be caught earlier as opposed to later. I wonder if it might be possible to point the CLI tool against the file whenever a module is saved, some plugins do this without taking any advantage of an LSP, for example https://github.com/InSyncWithFoo/pyright-for-pycharm (not VSCode, but the same idea)

### PR DESCRIPTION
**Pull request by caelean**
Fixes #57 

This uses the [VS Code python tool extension template](https://github.com/microsoft/vscode-python-tools-extension-template) from Microsoft. Most of the code is boilerplate; tach-specific stuff is in `lsp_server.py` and `tach_utils.py`.

Live and testable here: https://marketplace.visualstudio.com/items?itemName=Gauge.tach

### CODE REVIEW

File: python/tach/cache/access.py [hunk 1]
```diff
@@ -6,6 +6,8 @@
 from tach.cache.setup import resolve_dot_tach
 from tach.filesystem import find_project_config_root
 
+# clean_project()
+
```
Comment (emdoyle) [inline python/tach/cache/access.py:10]:
```suggestion
```

File: python/tach/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from __future__ import annotations
```
Comment (emdoyle) [inline python/tach/__init__.py:1]: Is this newly required/does this apply to all of our python code?

**Review by emdoyle — COMMENTED** [review]
Would be cool to avoid the manual `root` setting for initial installation if possible! Some other nits and questions about how it works, but will try to install it locally and try it out later tonight.

File: lsp/vscode/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1 @@
+update types
```
[the remark this answers is not in the document]
Comment (caelean) [reply lsp/vscode/CHANGELOG.md:1]: No, just needs to exist

File: python/tach/__init__.py (hunk 2 above)
Comment (caelean) [reply python/tach/__init__.py]: good catch, leftover from testing invalid imports

File: lsp/vscode/package.json [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    "displayName": "Tach",
+    "description": "Linting support for python files using `tach`.",
+    "version": "0.3.1",
+    "preview": true,
+    "serverInfo": {
+        "name": "Tach",
+        "module": "tach"
+    },
+    "publisher": "gauge",
+    "license": "LGPL-3.0-only",
```
[the remark this answers is not in the document]
Comment (caelean) [reply lsp/vscode/package.json:12]: Yep

File: Makefile [hunk 5]
```diff
@@ -28,46 +30,72 @@ deps: ## Install dependencies
...
+		uv venv $(VENV); \
+		echo "Virtual environment created at $(VENV)"; \
+	else \
+		echo "Virtual environment already exists at $(VENV)"; \
+	fi
+
+	source $(VENV_BIN)/activate && \
+	uv pip install -r dev-requirements.txt
+    # Note: this can take up to 10 minutes
+    cd lsp/vscode && nox --session setup
```
[the remark this answers is not in the document]
Comment (caelean) [reply Makefile:48]: yikes looks like this is a tabs vs spaces dif - might convert the whole file to spaces?

**Comment by emdoyle** [comment]
When I installed from the marketplace link, I hit an issue that seems to indicate the Rust extension wasn't compiled into the bundle:

image

Also will we publish this through GHA? If I understand the lsp.yml correctly it's just for CI right?

File: pyproject.toml [hunk 6]
```diff
@@ -61,7 +61,7 @@ runtime-evaluated-decorators = [
 exempt-modules = ["typing", "typing_extensions"]
 
 [tool.pyright]
-include = ["python/tach"]
+include = ["python/tach", "lsp/vscode/src/test/python_tests", "lsp/vscode/bundled/tool"]
```
[the remark this answers is not in the document]
Comment (caelean) [reply pyproject.toml:64]: Going to split out from this pr for now

**Review by emdoyle — APPROVED** [review]
LGTM! Works when installing from the VSCode Marketplace.

**Pull request merged**